app/src/main/java/de/danoeh/antennapod/activity/VideoplayerActivity.java

```diff
@@ -480,4 +484,95 @@ void stop() {
+        AudioManager audioManager = (AudioManager) getSystemService(AUDIO_SERVICE);
+        Float videoPercent = null;
+
+        switch (keyCode) {
+            case KeyEvent.KEYCODE_ENTER: //Fallthrough
+            case KeyEvent.KEYCODE_SPACE:
+                //Play/Pause
+                onPlayPause();
+                toggleVideoControlsVisibility();
+                break;
+            case KeyEvent.KEYCODE_DPAD_LEFT:
+                //Go Back
+                onRewind();
+                showSkipAnimation(false);
+                break;
+            case KeyEvent.KEYCODE_DPAD_RIGHT:
+                //Go Forward
+                onFastForward();
+                showSkipAnimation(true);
+                break;
+            case KeyEvent.KEYCODE_M:
+                //Mute/Unmute
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    audioManager.adjustStreamVolume(AudioManager.STREAM_MUSIC, AudioManager.ADJUST_TOGGLE_MUTE, 0);
+                } else {
+                    return super.onKeyUp(keyCode, event);
+                }
+                break;
+            case KeyEvent.KEYCODE_DPAD_UP:
+                //Raise volume
+                audioManager.adjustStreamVolume(AudioManager.STREAM_MUSIC,
+                        AudioManager.ADJUST_RAISE, AudioManager.FLAG_SHOW_UI);
+                break;
+            case KeyEvent.KEYCODE_DPAD_DOWN:
+                //Raise volume
+                audioManager.adjustStreamVolume(AudioManager.STREAM_MUSIC,
+                        AudioManager.ADJUST_LOWER, AudioManager.FLAG_SHOW_UI);
+                break;
+            case KeyEvent.KEYCODE_ESCAPE:
+            case KeyEvent.KEYCODE_F:
+                //Exit fullscreen mode
+                onBackPressed();
+                break;
+            case KeyEvent.KEYCODE_P:
+                //Toggle picture-in-picture mode
+                compatEnterPictureInPicture();
+                break;
+            //Go to x% of video:
+            case KeyEvent.KEYCODE_0:
+                videoPercent = 0f;
+                break;
+            case KeyEvent.KEYCODE_1:
+                videoPercent = 0.1f;
+                break;
+            case KeyEvent.KEYCODE_2:
+                videoPercent = 0.2f;
+                break;
+            case KeyEvent.KEYCODE_3:
+                videoPercent = 0.3f;
+                break;
+            case KeyEvent.KEYCODE_4:
+                videoPercent = 0.4f;
+                break;
+            case KeyEvent.KEYCODE_5:
+                videoPercent = 0.5f;
+                break;
+            case KeyEvent.KEYCODE_6:
+                videoPercent = 0.6f;
+                break;
+            case KeyEvent.KEYCODE_7:
+                videoPercent = 0.7f;
+                break;
+            case KeyEvent.KEYCODE_8:
+                videoPercent = 0.8f;
+                break;
+            case KeyEvent.KEYCODE_9:
+                videoPercent = 0.9f;
+                break;
```

Comment:
Something like the following would make the code shorter and also allows to remove the variable `videoPercent`. I think having `super.onKeyUp(keyCode, event)` in the end of the method would also make it easier to see that the method still calls the system method.

This is a pretty common pattern when converting chars in `c/c++`, so I guess it should be okay to understand (even if it is a bit less self-explanatory than your version).

```java
if (keyCode >= KeyEvent.KEYCODE_0 && keyCode <= KeyEvent.KEYCODE_9) {
    controller.seekTo((int) (0.1f * (keyCode - KeyEvent.KEYCODE_0) * controller.getDuration()));
    return true;
}
```